bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/ircc/IrccProtocol.java

```diff
@@ -0,0 +1,823 @@
+                        UnDefType.UNDEF);
+                callback.stateChanged(SonyUtil.createChannelId(IrccConstants.GRP_VIEWING, IrccConstants.CHANNEL_TITLE),
+                        UnDefType.UNDEF);
+                callback.stateChanged(
+                        SonyUtil.createChannelId(IrccConstants.GRP_VIEWING, IrccConstants.CHANNEL_DURATION),
+                        UnDefType.UNDEF);
+
+                if (irccStatusList.isDisk()) {
+                    refreshContentInformation();
+                }
+
+            } else {
+                if (!isViewing.getAndSet(true)) {
+                    refreshIsViewing();
+                }
+                final String id = viewing.getItemValue(IrccStatusItem.ID);
+                final String source = viewing.getItemValue(IrccStatusItem.SOURCE);
+                final String source2 = viewing.getItemValue(IrccStatusItem.SOURCE2);
+                final String clazz = viewing.getItemValue(IrccStatusItem.CLASS);
+                final String title = viewing.getItemValue(IrccStatusItem.TITLE);
+                final String dur = viewing.getItemValue(IrccStatusItem.DURATION);
+
+                callback.stateChanged(SonyUtil.createChannelId(IrccConstants.GRP_VIEWING, IrccConstants.CHANNEL_ID),
+                        id == null ? UnDefType.NULL : new StringType(id));
+                callback.stateChanged(SonyUtil.createChannelId(IrccConstants.GRP_VIEWING, IrccConstants.CHANNEL_SOURCE),
+                        source == null ? UnDefType.NULL : new StringType(source));
+                callback.stateChanged(
+                        SonyUtil.createChannelId(IrccConstants.GRP_VIEWING, IrccConstants.CHANNEL_SOURCE2),
+                        source == null ? UnDefType.NULL : new StringType(source2));
+                callback.stateChanged(SonyUtil.createChannelId(IrccConstants.GRP_VIEWING, IrccConstants.CHANNEL_CLASS),
+                        clazz == null ? UnDefType.NULL : new StringType(clazz));
+                callback.stateChanged(SonyUtil.createChannelId(IrccConstants.GRP_VIEWING, IrccConstants.CHANNEL_TITLE),
+                        title == null ? UnDefType.NULL : new StringType(title));
+                if (StringUtils.isEmpty(dur)) {
+                    callback.stateChanged(
+                            SonyUtil.createChannelId(IrccConstants.GRP_VIEWING, IrccConstants.CHANNEL_DURATION),
+                            UnDefType.NULL);
+                } else {
+                    try {
+                        callback.stateChanged(
+                                SonyUtil.createChannelId(IrccConstants.GRP_VIEWING, IrccConstants.CHANNEL_DURATION),
+                                new DecimalType(Integer.parseInt(dur)));
+                    } catch (final NumberFormatException e) {
+                        logger.error("Could not convert {} into an integer", dur);
```

Comment:
Shouldn't the state still be updated with something? an undef perhaps?

Reply:
Also, is this so catastrophic that it requires the logging level of ERROR?

Reply:
Hadn't really thought about that - but sounds fun!  Done